tt-train/sources/ttml/models/llama.cpp

```diff
@@ -185,18 +184,112 @@ Llama::Llama(const LlamaConfig& config) : m_config(config) {
     common::transformer::initialize_weights_gpt2(*this);
 }
 
-ttml::autograd::TensorPtr Llama::operator()(const ttml::autograd::TensorPtr& x, const ttml::autograd::TensorPtr& mask) {
+void Llama::initialize_kv_cache(const uint32_t batch_size) {
+    uint32_t head_dim = m_config.embedding_dim / m_config.num_heads;
+    uint32_t max_seq_len = m_config.max_sequence_length;
+    uint32_t num_groups = m_config.num_groups;
+
+    fmt::print("Initializing KV cache:\n");
+    fmt::print("    Batch size: {}\n", batch_size);
+    fmt::print("    Num layers: {}\n", m_config.num_blocks);
+    fmt::print("    Num groups: {}\n", num_groups);
+    fmt::print("    Max sequence length: {}\n", max_seq_len);
+    fmt::print("    Head dim: {}\n", head_dim);
+
+    m_kv_cache.clear();
+    m_kv_cache.reserve(m_config.num_blocks);
+
+    // Create cache tensors in DRAM for persistence across operations
+    // Shape: [batch_size, num_groups, max_seq_len, head_dim]
+    auto dram_memory_config = ttnn::MemoryConfig{ttnn::TensorMemoryLayout::INTERLEAVED, ttnn::BufferType::DRAM};
+
+    for (uint32_t layer_idx = 0; layer_idx < m_config.num_blocks; ++layer_idx) {
+        auto k_cache_shape = ttnn::Shape({batch_size, num_groups, max_seq_len, head_dim});
+        auto v_cache_shape = ttnn::Shape({batch_size, num_groups, max_seq_len, head_dim});
```

Comment:
consider this instead
```suggestion
        auto v_cache_shape = k_cache_shape;
```